### DISCUSSION TITLE
How can I use a proposed-API extension on github.dev?

### DISCUSSION THREAD

**Discussion by gjsjohnmurray** [microsoft/vscode-discussions#445] [Extension Development QnA] [answered]
https://code.visualstudio.com/api/advanced-topics/using-proposed-api#sharing-extensions-using-the-proposed-api documents two ways to tell VS Code Desktop that it can allow an extension to use proposed API.
- add `--enable-proposed-api=<YOUR-EXTENSION-ID>` when launching the executable
- use the `Preferences: Configure Runtime Arguments` command to edit `argv.json`

But I don't know how to do either of these when running in a GitHub Codespace.

Any ideas?

**Comment by alefragnani** [comment]
I guess this is not supported right now, because both GitHub Codespaces and vscode.dev are not _your_ VS Code installations, and the ways described in the docs are touching on the ways how VS Code starts. 

But, if `--enable-proposed-api` could be loaded from environment variables (instead of an arg to the executable), I think this could be an alternative to Codespaces, because you could define it in the `remoteEnv` entry at `devcontainer.json`.

Vscode.dev, on the other hand, would require a different approach.

I wonder why this can't be a regular user setting, or simplify the process and remove such requirement at all.

**Comment by r3m0t** [comment]
Did you try `Insiders.vscode.dev`?

**Comment by isidorn** [comment, marked as answer]
Maybe @joyceerhl knows

**Reply by joyceerhl** [reply]
There is no way to do this now, this would be a feature request. cc @sandy081

**Reply by sandy081** [reply]
CC @jrieken
